wire-library/wire-schema/src/jvmMain/kotlin/com/squareup/wire/schema/SchemaLoader.kt

```diff
@@ -150,14 +151,27 @@ class SchemaLoader {
         throw FileNotFoundException("Failed to locate $proto in $sources")
       }
 
-      loaded[proto] = ProtoFile.get(element)
+      val protoFile = ProtoFile.get(element)
+      if (protoFile.syntax == ProtoFile.Syntax.PROTO_3) {
+        skippedForSyntax += protoFile
+        continue
+      }
+
+      loaded[proto] = protoFile
 
       // Queue dependencies to be loaded.
       for (importPath in element.imports) {
         protos.addLast(importPath)
       }
     }
 
+    if (skippedForSyntax.isNotEmpty()) {
+      println("""Skipped .proto files with unsupported syntax. Add this line to fix:
```

Comment:
I don't think this class should directly print.

Reply:
It's a best effort to avoid crashing. I can relook tomorrow if I can come up with a good cheap idea.

Reply:
We have no problem _parsing_ proto3. I'd rather see the callers fail if possible by looking into the schema. There will be duplication, but the problem is codegen and not parsing.